cmd/cluster-capi-operator/main.go

```diff
@@ -243,79 +268,88 @@ func main() {
 	}
 }
 
-func getClusterOperatorStatusClient(mgr manager.Manager, controller string, managedNamespace string) operatorstatus.ClusterOperatorStatusClient {
+func getClusterOperatorStatusClient(mgr manager.Manager, currentFeatureGates featuregates.FeatureGate, controller string, managedNamespace string) operatorstatus.ClusterOperatorStatusClient {
 	return operatorstatus.ClusterOperatorStatusClient{
 		Client:           mgr.GetClient(),
 		Recorder:         mgr.GetEventRecorderFor(controller),
 		ReleaseVersion:   util.GetReleaseVersion(),
 		ManagedNamespace: managedNamespace,
+		FeatureGates:     currentFeatureGates,
 	}
 }
 
-func setupPlatformReconcilers(mgr manager.Manager, infra *configv1.Infrastructure, platform configv1.PlatformType, containerImages map[string]string, applyClient *kubernetes.Clientset, apiextensionsClient *apiextensionsclient.Clientset, managedNamespace string) {
+func setupPlatformReconcilers(mgr manager.Manager, infra *configv1.Infrastructure, platform configv1.PlatformType, containerImages map[string]string, applyClient *kubernetes.Clientset,
+	apiextensionsClient *apiextensionsclient.Clientset, currentFeatureGates featuregates.FeatureGate, managedNamespace, sourceNamespace string) {
 	// Only setup reconcile controllers and webhooks when the platform is supported.
 	// This avoids unnecessary CAPI providers discovery, installs and reconciles when the platform is not supported.
 	switch platform {
 	case configv1.AWSPlatformType:
-		setupReconcilers(mgr, infra, platform, &awsv1.AWSCluster{}, containerImages, applyClient, apiextensionsClient, managedNamespace)
+		setupReconcilers(mgr, infra, platform, &awsv1.AWSCluster{}, containerImages, applyClient, apiextensionsClient, currentFeatureGates, managedNamespace, sourceNamespace)
 		setupWebhooks(mgr)
 	case configv1.GCPPlatformType:
-		setupReconcilers(mgr, infra, platform, &gcpv1.GCPCluster{}, containerImages, applyClient, apiextensionsClient, managedNamespace)
+		setupReconcilers(mgr, infra, platform, &gcpv1.GCPCluster{}, containerImages, applyClient, apiextensionsClient, currentFeatureGates, managedNamespace, sourceNamespace)
 		setupWebhooks(mgr)
 	case configv1.AzurePlatformType:
 		azureCloudEnvironment := getAzureCloudEnvironment(infra.Status.PlatformStatus)
 		if azureCloudEnvironment == configv1.AzureStackCloud {
 			klog.Infof("Detected Azure Cloud Environment %q on platform %q is not supported, skipping capi controllers setup", azureCloudEnvironment, platform)
-			setupUnsupportedController(mgr, managedNamespace)
+			setupClusterOperatorController(mgr, managedNamespace, currentFeatureGates, true)
```

Comment:
Would it make sense to set up the clusteroperator controller before the switch, and remove it from `setupReconcilers`? Then we don't need this else, or the default case down below?